developer-workflow/c-api.rst

```diff
@@ -122,6 +122,32 @@ Guidelines for expanding/changing the public API
   - ``return 0``: lookup succeeded; no item was found
   - ``return 1``: lookup succeeded; item was found
 
+- APIs with output parameters should ensure that each output parameter is
+  initialised for all code paths.
```

Comment:
Amerian spelling is initialize with a Z

Maybe mention that the doc doesn't have to document if outputs are initialiazed on error case? But must be initialized for "lesser" case?